addons/easy_charts/Utilities/Scripts/ScatterChartBase.gd

```diff
@@ -0,0 +1,335 @@
+		"hint_string": "0.001, 10",
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Display/x_decim",
+		"type": TYPE_REAL
+	},
+	{
+		"hint": PROPERTY_HINT_RANGE,
+		"hint_string": "0.001, 10",
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Display/y_decim",
+		"type": TYPE_REAL
+	},
+	
+	# Chart Style
+	{ 
+		"hint": 24, 
+		"hint_string": ("%d/%d:%s"
+		%[TYPE_INT, PROPERTY_HINT_ENUM,
+		PoolStringArray(PointShapes.keys()).join(",")]),
+		"name": "Chart_Style/points_shape", 
+		"type": TYPE_ARRAY, 
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/function_colors",
+		"type": TYPE_COLOR_ARRAY
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/box_color",
+		"type": TYPE_COLOR
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/v_lines_color",
+		"type": TYPE_COLOR
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/h_lines_color",
+		"type": TYPE_COLOR
+	},
+	{
+		"class_name": "Font",
+		"hint": PROPERTY_HINT_RESOURCE_TYPE,
+		"hint_string": "Font",
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/font",
+		"type": TYPE_OBJECT
+	},
+	{
+		"class_name": "Font",
+		"hint": PROPERTY_HINT_RESOURCE_TYPE,
+		"hint_string": "Font",
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/bold_font",
+		"type": TYPE_OBJECT
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/font_color",
+		"type": TYPE_COLOR
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/use_template",
+		"type": TYPE_BOOL
+	},
+	{
+		"hint": PROPERTY_HINT_ENUM,
+		"hint_string": PoolStringArray(Utilities.templates.keys()).join(","),
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Style/template",
+		"type": TYPE_INT
+	},
+	
+	# Chart Modifiers
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Modifiers/treshold",
+		"type": TYPE_VECTOR2
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Modifiers/only_disp_values",
+		"type": TYPE_VECTOR2
+	},
+	{
+		"hint": PROPERTY_HINT_NONE,
+		"usage": PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE,
+		"name": "Chart_Modifiers/invert_chart",
+		"type": TYPE_BOOL
+	},
+	]
+
+
+func _get_property_list():
+	return property_list
```

Comment:
These two lines could be removed, since the function will be called on inherited children to customize the property list